### PR TITLE
Shardstore Unittest Fix

### PR DESCRIPTION
**Pull request by undertome**
## High Level Overview of Change

This fixes some recent failures in the shard database unit tests.

### Context of Change

The unit test functions `testDeterministicShard` and `testImportWithOnlineDelete` were modified to fix test case failures. Please limit the review to 7778710 as this fix is stacked onto several unmerged PRs.

### Type of Change

- [x ] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

**Comment by scottschurr** [comment]
:tada:

File: src/test/nodestore/DatabaseShard_test.cpp [hunk 1]
```diff
@@ -1140,6 +1135,20 @@ class DatabaseShard_test : public TestBase
...
+
+            auto static const ripemd160Key =
+                ripemd160File((path / "nudb.key").string());
+            auto static const ripemd160Dat =
+                ripemd160File((path / "nudb.dat").string());
+
+            BEAST_EXPECT(
+                ripemd160File((path / "nudb.key").string()) == ripemd160Key);
+            BEAST_EXPECT(
+                ripemd160File((path / "nudb.dat").string()) == ripemd160Dat);
```
Comment (miguelportilla) [inline src/test/nodestore/DatabaseShard_test.cpp:1150]: I don't think these two checks are necessary here as we perform them at the end of the loop.

File: src/test/nodestore/DatabaseShard_test.cpp [hunk 2]
```diff
@@ -1140,6 +1135,20 @@ class DatabaseShard_test : public TestBase
                 if (createShard(data, *db) < 0)
                     return;
             }
+
```
Comment (miguelportilla) [inline src/test/nodestore/DatabaseShard_test.cpp:1138]: The createShard call  should be
```if (!BEAST_EXPECT(createShard(data, *db) != std::nullopt))
                    return;```

File: src/test/nodestore/DatabaseShard_test.cpp [hunk 3]
```diff
@@ -1140,6 +1135,20 @@ class DatabaseShard_test : public TestBase
...
+            auto static const ripemd160Key =
+                ripemd160File((path / "nudb.key").string());
+            auto static const ripemd160Dat =
+                ripemd160File((path / "nudb.dat").string());
+
+            BEAST_EXPECT(
+                ripemd160File((path / "nudb.key").string()) == ripemd160Key);
+            BEAST_EXPECT(
+                ripemd160File((path / "nudb.dat").string()) == ripemd160Dat);
+
```
Comment (miguelportilla) [inline src/test/nodestore/DatabaseShard_test.cpp:1151]: The waitShard call below should be
```
if (!BEAST_EXPECT(waitShard(*db, 1) != std::nullopt))
    return;
```

File: src/test/nodestore/DatabaseShard_test.cpp [hunk 4]
```diff
@@ -1246,22 +1254,28 @@ class DatabaseShard_test : public TestBase
             Database& ndb = env.app().getNodeStore();
             BEAST_EXPECT(db);
 
-            auto const shardCount = 4;
+            // Create some ledgers for the shard store to
+            // import
```
Comment (miguelportilla) [inline src/test/nodestore/DatabaseShard_test.cpp:1258]: Nit. This comment can be a single line

File: src/test/nodestore/DatabaseShard_test.cpp [hunk 5]
```diff
@@ -1246,22 +1254,28 @@ class DatabaseShard_test : public TestBase
...
             {
                 // Wait until the import starts
-                std::this_thread::sleep_for(std::chrono::milliseconds(100));
+                std::this_thread::sleep_for(std::chrono::milliseconds(1));
             }
 
-            auto& store = env.app().getSHAMapStore();
-            auto lastRotated = store.getLastRotated();
+            // Enable online deletion now that the import has
+            // started
```
Comment (miguelportilla) [inline src/test/nodestore/DatabaseShard_test.cpp:1277]: Nit. This comment can be a single line

File: src/test/nodestore/DatabaseShard_test.cpp [hunk 6]
```diff
@@ -1273,6 +1287,7 @@ class DatabaseShard_test : public TestBase
                     {
                         // A rotation occurred during shard import. Not
                         // necessarily an error
+
```
Comment (miguelportilla) [inline src/test/nodestore/DatabaseShard_test.cpp:1290]: Nit: sequence can be renamed to ledgerSeq

File: src/test/nodestore/DatabaseShard_test.cpp [hunk 7]
```diff
@@ -1281,7 +1296,8 @@ class DatabaseShard_test : public TestBase
                 }
             });
 
-            data = TestData(seedValue * 2, 4, shardCount);
+            // Create more ledgers to trigger online deletion
+            data = TestData(seedValue * 2, 4, 1);
```
Comment (miguelportilla) [inline src/test/nodestore/DatabaseShard_test.cpp:1300]: Due to the defaults can also be
```
TestData(seedValue * 2);
```

**Review by miguelportilla — APPROVED** [review]
I left some nit comments but otherwise looks good. 👍

File: src/test/nodestore/DatabaseShard_test.cpp (hunk 1 above)
Comment (undertome) [reply src/test/nodestore/DatabaseShard_test.cpp:1150]: Done

File: src/test/nodestore/DatabaseShard_test.cpp (hunk 2 above)
Comment (undertome) [reply src/test/nodestore/DatabaseShard_test.cpp:1138]: Done

File: src/test/nodestore/DatabaseShard_test.cpp (hunk 3 above)
Comment (undertome) [reply src/test/nodestore/DatabaseShard_test.cpp:1151]: Done

File: src/test/nodestore/DatabaseShard_test.cpp (hunk 4 above)
Comment (undertome) [reply src/test/nodestore/DatabaseShard_test.cpp:1258]: Done

File: src/test/nodestore/DatabaseShard_test.cpp (hunk 5 above)
Comment (undertome) [reply src/test/nodestore/DatabaseShard_test.cpp:1277]: Done

File: src/test/nodestore/DatabaseShard_test.cpp (hunk 6 above)
Comment (undertome) [reply src/test/nodestore/DatabaseShard_test.cpp:1290]: Done

File: src/test/nodestore/DatabaseShard_test.cpp (hunk 7 above)
Comment (undertome) [reply src/test/nodestore/DatabaseShard_test.cpp:1300]: Good point - done

**Review by ximinez — COMMENTED** [review]
I know this is already merged, but I did look through it and everything looks good.

**Pull request closed without merging**